api/mfx_dispatch/linux/device_ids.h

```diff
@@ -415,19 +415,19 @@ std::vector <Device> get_devices() {
         Device device;
         std::string path = std::string(dir) + "/renderD" + std::to_string(128 + i) + vendor_id_file;
         FILE *file = fopen(path.c_str(), "r");
-        if (!file) break;
+        if (!file) continue;
         err = fscanf(file, "%x", &device.vendor_id);
         fclose(file);
-        if (err == EOF) break;
+        if (err == EOF) continue;
         if (device.vendor_id != 0x8086) {  // Filter out non-Intel devices
             continue;
         }
         path = std::string(dir) + "/renderD" + std::to_string(128 + i) + device_id_file;
         file = fopen(path.c_str(), "r");
-        if (!file) break;
+        if (!file) continue;
         err = fscanf(file, "%x", &device.device_id);
         fclose(file);
-        if (err == EOF) break;
+        if (err == EOF) continue;
```

Comment:
This still won't work for the docker use case I have in mind. If system has 2 gpus and user explicit adds only one with  `docker --device /dev/dri/renderD129` as an example, we will still have an issue because docker will have full access to sysfs entries, i.e. /sys/class/drm/* which we actually parse, but it will map only /dev/dri/renderD129 which is what application opens. So, we will populate the table with something which is not actually accessible, i.e. with /dev/dri/renderD128.

I still think that this change makes a sense, but we need to slightly modify it with something like:
```
    for (; i < 64; ++i) {
        Device device;
        std::string path = std::string(dir) + "/renderD" + std::to_string(128 + i) + vendor_id_file;
        FILE *file = fopen(path.c_str(), "r");
        if (!file) continue;
        err = fscanf(file, "%x", &device.vendor_id);
        fclose(file);
        if (err == EOF) continue;
        if (device.vendor_id != 0x8086) {  // Filter out non-Intel devices
            continue;
        }
        path = std::string(dir) + "/renderD" + std::to_string(128 + i) + device_id_file;
        file = fopen(path.c_str(), "r");
        if (!file) continue;
        err = fscanf(file, "%x", &device.device_id);
        fclose(file);
        if (err == EOF) continue;

        path = "/dev/dri/" + "/renderD" + std::to_string(128 + i);       <<<<<<<<<<<<<<<<<<<< NEW
        int fd=open(path.c_str());
        if (fd < 0) continue; //device not accessible

        device.platform = get_platform(device.device_id);
        result.emplace_back(device);
```

Reply:
Thank you! Change adjusted as  suggested.